corehq/apps/locations/migrations/0009_populate_location_type_from_string.py

```diff
@@ -0,0 +1,150 @@
+# encoding: utf-8
+from south.v2 import DataMigration
+from corehq.apps.domain.models import Domain
+
+
+EXCLUDE_DOMAINS = (
+    "drewpsi",
+    "psi",
+    "psi-ors",
+    "psi-test",
+    "psi-test2",
+    "psi-test3",
+    "psi-unicef",
+    "psi-unicef-wb",
+)
+
+
+def _couch_parent_loc(parent_code, couch_loc_types):
+    for loc_type in couch_loc_types:
+        if loc_type.get('code') == parent_code or loc_type['name'] == parent_code:
+            return loc_type
+    raise ValueError("Parent loc type {} not found on {}"
+                     .format(parent_code, domain))
+
+
+class Migration(DataMigration):
+
+    def make_loc_types(self, couch_loc_types, domain):
+
+        def get_or_create(couch_loc_type):
+            parents = couch_loc_type['allowed_parents']
+            if len(parents) != 1:
+                raise ValueError("Improperly configured location types on {}"
+                                 .format(domain))
+            elif parents[0] == '':
+                parent_type = None
+            else:
+                parent_type = get_or_create(
+                    _couch_parent_loc(parents[0], couch_loc_types)
+                )
+
+            return self.orm.LocationType.objects.get_or_create(
+                domain=domain,
+                code=couch_loc_type.get('code', couch_loc_type['name']),
+                defaults={
+                    'name': couch_loc_type['name'],
+                    'parent_type': parent_type,
+                    'administrative': couch_loc_type['administrative'] or False,
```

Comment:
am I missing something or do we need to add `shares_cases` and `view_descendants` here?